rerun_cpp/src/rerun/collection.hpp

```diff
@@ -355,6 +355,15 @@ namespace rerun {
             return data() + size();
         }
 
+        /// Returns a pointer to the first element, if any, else `nullptr`.
+        const TElement* first() const {
+            if (empty()) {
+                return nullptr;
+            } else {
+                return begin();
+            }
+        }
```

Comment:
`first` is very unidiomatic in C++, practically unheard of afaik! So I'd rather not add that, seems quite confusing
However, I think we should just make all this safer & better by having `data` return `nullptr` if the collection is empty. That way all of `data`/`begin`/`end` get better.
This also still lines up with what folks would expect from `begin` - the only condition really is that for an empty _thing_ `begin` == `end`. There doesn't seem to be much definition on what it is supposed to be usually

(plz extend existing (?) unit tests for empty collections accordingly)

Reply:
In the end I never made use of it, so I'll just delete it.